pkg/validations/cluster.go

```diff
@@ -215,3 +216,19 @@ func ValidateK8s128Support(clusterSpec *cluster.Spec) error {
 	}
 	return nil
 }
+
+// ValidateTemplateNameWithKubernetesVersion check if the template name in the machineconfig field includes kubernetes version.
+func ValidateTemplateNameWithKubernetesVersion(spec cluster.Spec) error {
```

Comment:
I don't think this validation belongs here. IMO it should be in cloudstack package